python-package/lightgbm/basic.py

```diff
@@ -1979,11 +1979,13 @@ def __init__(self, params=None, train_set=None, model_file=None, model_str=None,
                                      listen_time_out=params.get("listen_time_out", 120),
                                      num_machines=params.setdefault("num_machines", num_machines))
                     break
-            # construct booster object
-            train_set.construct()
             # copy the parameters from train_set
-            params.update(train_set.get_params())
+            if train_set.get_params():
+                params.update(train_set.get_params())
             params_str = param_dict_to_str(params)
+            train_set._update_params(params)
+            # construct booster object
+            train_set.construct()
```

Comment:
why about change is needed? also cc @StrikerRUS

Reply:
is `linear_tree` parameter needed in both the dataset and booster? I remember we fix that problem in https://github.com/microsoft/LightGBM/pull/2594 , and you change this accordingly.

Reply:
I've now updated `get_params` to include `linear_tree` so this change is not needed anymore.